activerecord/lib/active_record/model_schema.rb

```diff
@@ -484,6 +484,9 @@ def load_schema
         end
 
         def load_schema!
+          unless table_name
+            raise ActiveRecord::NoTableError, "#{self} has no table configured"
```

Comment:
I would make this error more actionable by stating what they need to do to set the table name eg. `"#{self} has no table configured. Set one with #{self}.table_name="`.